lib/slashdeploy/service.rb

```diff
@@ -95,6 +95,58 @@ def lock_environment(user, environment, options = {})
         stolen: stolen
     end
 
+    # Attempts to queue the user for the environment lock.
+    # Throws an error if there is no active lock for the environment.
+    #
+    # user - A User to own the lock
+    # environment - An Environment to lock.
+    # options     - A hash of options.
+    #               :message - An optional message.
+    #
+    # Returns a waiting Lock or Nil, if the user is already queued.
+    def queue_user_for_environment(user, environment, options = {})
```

Comment:
Part of me feels like this should be part of `SlashDeploy::Service#lock_environment`. If I'm enqueueing myself for a lock, and no one has the lock, then I should just get the lock. If it's locked by someone else, then I should just get enqueued (maybe make it an option, like `enqueue_if_locked: true`).

Same for `dequeue_user_for_environment`, maybe it should just use `unlock_environment` to release all active/waiting locks on an environment.

Thoughts? I could go either way if separate methods seems easier.